src/runtime_src/core/pcie/driver/linux/xocl/xocl_xgq.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * A GEM style device manager for PCIe based OpenCL accelerators.
+ *
+ * Copyright (C) 2021 Xilinx, Inc. All rights reserved.
+ *
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <linux/slab.h>
+#include <linux/spinlock.h>
+#include "xocl_xgq.h"
+#include "xgq_xocl_plat.h"
+
+#define CLIENT_ID_BITS 7
+#define MAX_CLIENTS (1 << CLIENT_ID_BITS)
+
+static uint16_t xocl_xgq_cid;
+
+struct xocl_xgq_client {
+	void			*xxc_client;
+	u32			 xxc_num_cmds;
+};
+
+struct xocl_xgq {
```

Comment:
same name being used in xgq.c. We need some better name for mgmt(vmr) and user(ert).
what about xocl_vmr_xgq and xocl_ert_xgq? or any other suggestions?